sql/core/src/test/scala/org/apache/spark/sql/execution/ExchangeSuite.scala

```diff
@@ -132,4 +135,33 @@ class ExchangeSuite extends SparkPlanTest with SharedSQLContext {
     val projection2 = cached.select("_1", "_3").queryExecution.executedPlan
     assert(!projection1.sameResult(projection2))
   }
+
+  test("SPARK-26601: Make broadcast-exchange thread pool configurable") {
+    val previousNumber = SparkSession.getActiveSession.get.sparkContext.conf
+      .get(StaticSQLConf.BROADCAST_EXCHANGE_MAX_THREAD_THREASHOLD)
+
+    SparkSession.getActiveSession.get.sparkContext.conf.
+      set(StaticSQLConf.BROADCAST_EXCHANGE_MAX_THREAD_THREASHOLD, 1)
+    assert(SQLConf.get.getConf(StaticSQLConf.BROADCAST_EXCHANGE_MAX_THREAD_THREASHOLD) === 1)
+
+    Future {
+      Thread.sleep(5*1000)
+    } (BroadcastExchangeExec.executionContext)
+
+    val f = Future {} (BroadcastExchangeExec.executionContext)
+    intercept[TimeoutException] {
+      ThreadUtils.awaitResult(f, 3 seconds)
+    }
+
+    var executed = false
+    val ef = Future {
+      executed = true
+    } (BroadcastExchangeExec.executionContext)
+    ThreadUtils.awaitResult(ef, 3 seconds)
+    assert(executed)
+
+    // for other test
+    SparkSession.getActiveSession.get.sparkContext.conf.
+      set(StaticSQLConf.BROADCAST_EXCHANGE_MAX_THREAD_THREASHOLD, previousNumber)
+  }
```

Comment:
We need this test? In the other similar prs (e.g., https://github.com/apache/spark/pull/22847/files), we added no test. If necessary, we can add tests in `ExecutorSideSQLConfSuite` to check if executors can reference this static value...